pkg/karmadactl/deinit/deinit_test.go

```diff
@@ -0,0 +1,212 @@
+/*
+Copyright 2024 The Karmada Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deinit
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	fakeclientset "k8s.io/client-go/kubernetes/fake"
+
+	"github.com/karmada-io/karmada/pkg/util/names"
+)
+
+type resource struct {
+	name         string
+	resourceType string
+}
+
+func TestDeInitKarmada(t *testing.T) {
+	labels := map[string]string{karmadaBootstrappingLabelKey: "app-defaults"}
+	tests := []struct {
+		name       string
+		deInitOpts *CommandDeInitOption
+		resources  []resource
+		prep       func(client clientset.Interface, namespace string, resources []resource) error
+		verify     func(client clientset.Interface, namespace string, resources []resource) error
+		wantErr    bool
+	}{
+		{
+			name: "DeInitKarmada_DeInitKarmadaAfterItIsInitialized_KarmadaDeinit",
+			deInitOpts: &CommandDeInitOption{
+				KubeClientSet:  fakeclientset.NewSimpleClientset(),
```

Comment:
NewSimpleClientset has been deprecated, how about using `NewClientset` to replace it?